src/components/Indicator/Indicator.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import {
+  createShorthandFactory,
+  UIComponent,
+  UIComponentProps,
+  commonPropTypes,
+  ColorComponentProps,
+  customPropTypes,
+} from '../../lib'
+import { ReactProps, ShorthandValue } from '../../../types/utils'
+import Icon from '../Icon/Icon'
+
+export interface IndicatorProps extends UIComponentProps, ColorComponentProps {
+  /** The indicator can be direction in different directions. */
+  direction?: 'forward' | 'back' | 'top' | 'bottom'
+
+  /** The indicator can show specific icon if provided. */
+  icon?: ShorthandValue
+}
+
+/**
+ * An indicator is suggest additional content next to the element it is inside.
+ */
+class Indicator extends UIComponent<ReactProps<IndicatorProps>, any> {
+  static displayName = 'Indicator'
+
+  static create: Function
+
+  static className = 'ui-indicator'
+
+  static directionMap = {
+    forward: { unicode: '25B8', rotation: -90 },
+    back: { unicode: '25C2', rotation: 90 },
+    top: { unicode: '25B4', rotation: 180 },
+    bottom: { unicode: '25BE', rotation: 0 },
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({ children: false, content: false }),
+    direction: PropTypes.oneOf(['forward', 'back', 'top', 'bottom']),
+    icon: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    as: 'span',
+    direction: 'bottom',
+  }
+
+  renderComponent({ ElementType, classes, unhandledProps, rtl }) {
+    const { direction, icon, color } = this.props
+    const hexUnicode =
+      direction && Indicator.directionMap[this.getDirectionBasedOnRtl(rtl, direction)].unicode
+    const contentProps = !icon
+      ? {
+          dangerouslySetInnerHTML: {
+            __html: hexUnicode && this.isHex(hexUnicode) ? `&#x${hexUnicode};` : '',
+          },
+        }
+      : {
+          children: Icon.create(icon, {
+            defaultProps: { color, rotate: Indicator.directionMap[direction].rotation },
```

Comment:
Seems like this is making the assumption that all icons will always be rotated in the same way, is that right?

Reply:
I myself had doubts on this, but I don't see any other way how we can easily achieve this (from the other components it's not good practice to change some props for the icon as it is shorthand for another component - Indicator in this case). So we can either provide prop in the Indicator for rotating (similar that we have in the Icon), or we can have assumption that the indicator icon provided will always be the bottom one (and depending on the direction we can rotate it accordingly). I am open to other proposals as well.

Reply:
I would prefer something like this:
```jsx
overrideProps: ({rotate}) => ({rotate: Indicator.directionMap[direction].rotation + rotate})
```

Then I can use `send` icon as an indicator:
```jsx
<Indicator icon={{name: "send", rotate: 90}} direction="forward" />
```